SETUP/INSTALL.md

```diff
@@ -0,0 +1,389 @@
+
+### Unpack the DP code
+Somewhere within your web server's document hierarchy (possibly,
+though not necessarily, in the document root) unpack the DP code.
+This expands into a directory named 'c'.
+
+### Create other directories
+Create some additional directories under your web server's document
+hierarchy:
+
+    projects
+
+    d
+    d/locale
+    d/stats
+    d/stats/automodify_logs
+    d/teams
+    d/teams/avatar
+    d/teams/icon
+    d/pg
+    d/xmlfeeds
+    (You also need to put some files in d??)
+
+And create the following temporary directory to house WordCheck temp files:
+
+    /tmp/sp_check
+
+These should have permissions `drwxrwsr-x` and have the owner & group
+of the user that the DP code runs as (probably 'nobody' or 'www-data').
+
+### Create the directory for project file uploads
+In order to populate projects with images and initial text, the images
+and text must reside on the server. This can be done by users via the
+web with the built-in `remote_file_manager.php`, or via FTP/scp, or
+both. In either case, a directory outside the web space, but writeable
+by the web server, is required.
+
+If using `remote_file_manager.php`, you can create this directory wherever
+it makes sense to do so, although for security reasons it is strongly
+recommended that it be outside of your web space. Ensure the directory
+permissions and/or ownership allow the web server full access to the
+directory and its contents.
+
+If using FTP or scp, create a system user and have the uploads
+directory located inside that user's home directory. Ensure that
+new files and directories within this space have permissions set
+to 777 so that the web server has full access to them.
+
+See the 'Uploading and Creating Projects' section of `configuration.sh`
+for more details.
+
+### Configure the DP code (with site-specific settings)
+Make an editable copy of `c/SETUP/configuration.sh`, and put it outside
+your webserver's document hierarchy.
+
+Modify your copy of `configuration.sh` as appropriate. Details about each
+parameter are included in `configuration.sh`. Run:
+```bash
+c/SETUP/configure path-to-modified-configuration.sh path-to-code-dir
+```
+The configure script will use your modified `configuration.sh` to update
+the DP code with your site-specific settings.
+
+### Create the tables of the DP database
+```bash
+cd c/SETUP
+php -f install_db.php
+```
+
+### Move SETUP directory outside your webserver's document hierarchy
+The `c/SETUP/` directory is only needed for site configuration. **As a
+security measure, move it out of your webserver's document hierarchy.**
+Don't delete it as you may want to use its contents again to
+reconfigure the site.
+
+### At this point, your DP site should be functional.
+Try visiting it via your browser.
+(You may want to redirect/rewrite requests for foo/ to foo/c/.
+You could just create a foo/index.html that redirects.)
+Register as a new user.
+Of course, there aren't any projects to proofread yet.
```

Comment:
This doesn't seem to be a particularly useful statement. Does it even need to be there? If projects even need to be mentioned at this point, perhaps this should point to a document that describes creating a project? However, a bit down from here, defining a siteadmin is described; is that the purpose of registering this account?

Reply:
Yeah, I don't think that's necessary. I've removed it.